gnt4/docs/guides/opcode_group/09.md

```diff
@@ -2,68 +2,378 @@
+- [091C - ptr_bgez](#091C---ptr_bgez)
+- [091D - ptr_bl](#091D---ptr_bl)
+- [091E - ptr_beqzal](#091E---ptr_beqzal)
+- [091F - ptr_bnezal](#091F---ptr_bnezal)
+- [0920 - ptr_bgtzal](#0920---ptr_bgtzal)
+- [0921 - ptr_bgezal](#0921---ptr_bgezal)
+- [0922 - ptr_bltzal](#0922---ptr_bltzal)
+- [0923 - ptr_blezal](#0923---ptr_blezal)
+- [0924 - ptr_bltzal](#0924---ptr_bltzal)
+- [0925 - ptr_bgezal](#0925---ptr_bgezal)
+- [0926](#0926)
 
-## 0901
+## 0900 - ptr_debug
 
-## 0902
+Debug `sprintf` the given parameter using the print format `"0x%08x "`.
 
-## 0903
+## 0901 - ptr_mov
 
-## 0904
+Move pointer.
 
-## 0905
+- Two operands from `SEQ_RegCMD2`.
+- Stores result in the first operand.
+- Stores result in the conditional register `cr`.
 
-## 0906
+```c
+op1 = op2
+cr = op1
+```
 
-## 0907
+## 0902 - ptr_inc
 
-## 0908
+WARNING: This opcode is broken in GNT4, as it increments `sp` instead of the operand.
 
-## 0909
+Pointer increment.
 
-## 090A
+- One operand from `SEQ_RegCMD1`.
+- Stores result in the conditional register `cr`.
 
-## 090B
+```c
+sp++ // coding mistake by Eighting, should be op++
+cr = op1
+```
 
-## 090C
+## 0903 - ptr_dec
 
-## 0914
+Pointer decrement. This will decrement by the size of a word (4 bytes).
 
-## 0915
+- One operand from `SEQ_RegCMD1`.
+- Stores result in the operand.
+- Stores result in the conditional register `cr`.
 
-## 0916
+```c
+op1--
+cr = op1
+```
 
-## 0917
+## 0904 - ptr_add
 
-## 0918
+Pointer addition.
 
-## 0919
+- Two operands from `SEQ_RegCMD2`.
+- Stores result in the first operand.
+- Stores result in the conditional register `cr`.
 
-## 091A
+```c
+op1 = op1 + op2
+cr = op1
+```
 
-## 091B
+## 0905 - ptr_sub
 
-## 091C
+Pointer subtraction.
 
-## 091D
+- Two operands from `SEQ_RegCMD2`.
+- Stores result in the first operand.
+- Stores result in the conditional register `cr`.
 
-## 091E
+```c
+op1 = op1 - op2
+cr = op1
+```
 
-## 091F
+## 0906 - ptr_subc
 
-## 0920
+Pointer subtraction (compare).
 
-## 0920
+- Two operands from `SEQ_RegCMD2`.
+- Stores result in the conditional register `cr`.
 
-## 0921
+```c
+cr = op1 - op2
+```
 
-## 0922
+## 0907 - ptr_move
 
-## 0923
+Move pointer (ephemeral). Ephemeral in this case means it does not store to `cr`.
 
-## 0924
+- Two operands from `SEQ_RegCMD2`.
+- Stores result in the first operand.
 
-## 0925
+```c
+op1 = op2
+```
+
+## 0908 - ptr_from_offset
+
+Convert an SEQ offset to an SEQ pointer address.
+
+- Two operands from `SEQ_RegCMD2`.
+- Stores result in the first operand.
+
+```c
+op1 = SEQ_START_PTR + op2
+```
+
+## 0909 - ptr_to_offset
+
+Convert an SEQ pointer address to an SEQ offset.
+
+- Two operands from `SEQ_RegCMD2`.
+- Stores result in the first operand.
+
+```c
+op1 = op2 - SEQ_START_PTR
+```
+
+## 090A - ptr_push
+
+Push a pointer to the stack.
+
+- One operand from `SEQ_RegCMD1`.
+
+```c
+sp--
+sp = op1
+```
+
+## 090B - ptr_pop
+
+Pop a pointer from the stack.
+
+- One operand from `SEQ_RegCMD1`.
+
+```c
+op1 = sp
+sp++
+```
+
```

Comment:
No dereferencing of the stack pointer is a bit confusing when reading.

Making it 
```
sp--
*sp = op1
```
and
```
op1 = *sp 
sp++
```
Would make it easier to understand how it works.

Reply:
Good point. I've updated each of these in https://github.com/NicholasMoser/Naruto-GNT-Modding/pull/16/commits/ea4785a6fca0a0c6cb40c60b6ac833e1996d8ab7 as well as a few other places where the stack pointer is referenced.